ir/pass_manager.cpp

```diff
@@ -85,8 +85,8 @@ const IR::Node *PassManager::apply_visitor(const IR::Node *program, const char *
                     LOG3(log_indent << "heap after " << v->name() << ": in use " << n4(mem)
                                     << "B, max " << n4(maxmem) << "B");
                 }
-                if (stop_on_error && ::errorCount() > initial_error_count) break;
-                if ((program = after) == nullptr) break;
+                if (stop_on_error && ::errorCount() > initial_error_count) early_exit_flag = true;
+                if ((program = after) == nullptr) early_exit_flag = true;
```

Comment:
What does it mean when `(program = after) == nullptr`? Unclear to me.

Reply:
I think this is C/C++'s fun "you can embed assignment statements inside of expressions" thing, semantically equivalent to doing:

```
program = after;
if (program == nullptr) early_exit_flag = true;
```
Or maybe I don't understand your question.

Reply:
Oh yes, it was more meant as a clarification question under which circumstances the `after` is nullptr. 
But that expression should also be simplified while we are at it.

Reply:
At least the Transform's code suggests it is possible to return `nullptr` in the pre/postorder and if you do that for root the entire `apply` would yield `nullptr`. This would be quite weird I guess, but it is allowed.

I've pushed a commit that eliminates the `after` variable altogether.